src/common/connection.js

```diff
@@ -118,6 +126,10 @@ class Connection extends EventEmitter {
     });
   }
 
+  _onOpenError(reject, error) {
+    reject(new DisconnectedError(error && error.message));
```

Comment:
I think we probably want `NotConnectedError` here; `DisconnectedError` is for when it was connected but got disconnected, whereas `NotConnectedError` is for when it just wasn't connected in the first place.